specs-go/state.go

```diff
@@ -1,5 +1,20 @@
 package specs
 
+const (
+	// StateCreating indicates that the container is being created
+	StateCreating = "creating"
+
+	// StateCreated indicates that the runtime has finished the create operation
+	StateCreated = "created"
+
+	// StateRunning indicates that the container process has executed the
+	// user-specified program but has not exited
+	StateRunning = "running"
+
+	// StateStopped indicates that the container process has exited
+	StateStopped = "stopped"
+)
```

Comment:
Do we want these to have the same type (similar to https://godoc.org/github.com/docker/docker/api/types/mount#Consistency) ?

```suggestion
// ContainerState represents the state of a container.
type ContainerState string

const (
	// StateCreating indicates that the container is being created
	StateCreating ContainerState  = "creating"

	// StateCreated indicates that the runtime has finished the create operation
	StateCreated ContainerState  = "created"

	// StateRunning indicates that the container process has executed the
	// user-specified program but has not exited
	StateRunning ContainerState  = "running"

	// StateStopped indicates that the container process has exited
	StateStopped ContainerState  = "stopped"
)
```

Reply:
Yeah I agree, there should be a dedicated type for it.

Reply:
Now, should the `State.Status` field should also use this type, ie

```diff
// State holds information about the runtime state of the container.
type State struct {
...
         // Status is the runtime status of the container.
-        Status string `json:"status"`
+        Status ContainerState `json:"status"`
...
```

I am asking because of this https://github.com/opencontainers/runc/pull/2520/commits/34ac72f87cd41ea116828760bb70ed70960c8d8e#r456629563

Reply:
IMO that makes sense

Reply:
I think it also makes sense, existing hooks shouldn't be broken by this too so it would just be a matter of updating the existing runtimes.

I'll open a PR later today.

Reply:
It's here: https://github.com/opencontainers/runtime-spec/pull/1056

Reply:
Agreed.